alica_engine/include/engine/RunningPlan.h

```diff
@@ -115,17 +116,14 @@ class RunningPlan : public enable_shared_from_this<RunningPlan> {
     bool allChildrenStatus(PlanStatus ps);
     bool anyChildrenTaskSuccess();
     void activate();
-    EntryPoint* getActiveEntryPoint() const { return _activeEntryPoint; }
+    const EntryPoint* getActiveEntryPoint() const { return _activeEntryPoint; }
     void setActiveEntryPoint(EntryPoint* activeEntryPoint);
-    void limitToRobots(std::unordered_set<const supplementary::AgentID*, supplementary::AgentIDHash,
-            supplementary::AgentIDEqualsComparator>
-                    robots);
+    void limitToRobots(const AgentSet& robots);
     std::shared_ptr<CycleManager> getCycleManagement();
     void revokeAllConstraints();
     void attachPlanConstraints();
-    bool recursiveUpdateAssignment(list<shared_ptr<SimplePlanTree>> spts,
-            std::vector<const supplementary::AgentID*>& availableAgents, list<const supplementary::AgentID*> noUpdates,
-            AlicaTime now);
+    bool recursiveUpdateAssignment(list<shared_ptr<SimplePlanTree>> spts, AgentSet& availableAgents,
+            list<const supplementary::AgentID*> noUpdates, AlicaTime now);
```

Comment:
AgentList might be better

Reply:
No AgentList, lists are bad, we'll replace the datatypes one by one.